src/inventory_ui.cpp

```diff
@@ -846,66 +1261,71 @@ void inventory_compare_selector::toggle_entry( inventory_entry *entry )
     on_change( *entry );
 }
 
-inventory_drop_selector::inventory_drop_selector( player &u, const std::string &title, const item_location_filter &filter ) :
-    inventory_selector( u, title, filter ),
-    dropping()
-{
-    insert_selection_column( "ITEMS TO DROP", _( "ITEMS TO DROP" ) );
-}
+inventory_drop_selector::inventory_drop_selector( const player &p,
+        const inventory_selector_preset &preset ) :
+    inventory_multiselector( p, preset, _( "ITEMS TO DROP" ) ) {}
 
 std::list<std::pair<int, int>> inventory_drop_selector::execute()
 {
-    prepare_columns( true );
-
     int count = 0;
     while( true ) {
-        refresh_window();
+        update();
 
-        const std::string action = ctxt.handle_input();
-        const long ch = ctxt.get_raw_input().get_first_input();
-        auto entry = find_entry_by_invlet( ch );
+        const inventory_input input = get_input();
 
-        if( ch >= '0' && ch <= '9' ) {
+        if( input.ch >= '0' && input.ch <= '9' ) {
             count = std::min( count, INT_MAX / 10 - 10 );
             count *= 10;
-            count += ch - '0';
-        } else if( entry != nullptr ) {
-            set_drop_count( *entry, count );
+            count += input.ch - '0';
+        } else if( input.entry != nullptr ) {
+            set_drop_count( *input.entry, count );
             count = 0;
-        } else if( action == "RIGHT" ) {
+        } else if( input.action == "RIGHT" ) {
```

Comment:
Using `LEFT` for navigation but `RIGHT` for selection in a multi-column display is a bad UI design (#17427). This needs fixing in this PR rather than further propagating.